src/main/java/frc/robot/subsystems/hood/HoodIOSim.java

```diff
@@ -0,0 +1,59 @@
+package frc.robot.subsystems.hood;
+
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VelocityVoltage;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.Timer;
+import lib.motors.TalonFXSim;
+
+public class HoodIOSim implements HoodIO {
+    private final TalonFXSim motor;
+
+    private final PositionVoltage control = new PositionVoltage(0);
+    private final SimpleMotorFeedforward motorFeedforward;
+
+    public HoodIOSim() {
+        motor = new TalonFXSim(1, HoodConstants.GEAR_RATIO, HoodConstants.MOMENT_OF_INERTIA, 1);
+
+        motor.setController(
+                new PIDController(
+                        HoodConstants.kP.get(), HoodConstants.kI.get(), HoodConstants.kD.get()));
+
+        motorFeedforward =
+                new SimpleMotorFeedforward(
+                        HoodConstants.kS.get(), HoodConstants.kV.get(), HoodConstants.kA.get());
+    }
+
+    @Override
+    public void updateInternalEncoder() {}
+
+    @Override
+    public void setAngle(MutableMeasure<Angle> angle) {
+        inputs.controlMode = Mode.ANGLE;
+        inputs.angleSetpoint = angle.mut_replace(angle);
+        motor.setControl(
+                control.withPosition(angle.in(Units.Radians))
+                        .withFeedForward(
+                                motorFeedforward.calculate(
+                                        angle.in(Units.RotationsPerSecond.getUnit()))));
+    }
```

Comment:
If you have five consecutive parentheses (`)))))`), you should probably extract something into a variable.

Reply:
ad207d2